apps/condo/domains/user/schema/SigninResidentUserService.js

```
@@ -0,0 +1,90 @@
+} = require('@condo/domains/user/constants/errors')
+const { ConfirmPhoneAction: ConfirmPhoneActionServerUtils, User: UserServerUtils } = require('@condo/domains/user/utils/serverSchema')
+const { getRandomString } = require('@core/keystone/test.utils')
+
+const SigninResidentUserService = new GQLCustomSchema('SigninResidentUserService', {
+    types: [
+        {
+            access: true,
+            type: 'input SigninResidentUserInput { dv: Int!, sender: JSON!, token: String! }',
+        },
+        {
+            access: true,
+            type: 'type SigninResidentUserOutput { user: User, token: String! }',
+        },
+    ],
+
+    mutations: [
+        {
+            access: access.canSigninResidentUser,
+            schema: 'signinResidentUser(data: SigninResidentUserInput!): SigninResidentUserOutput',
+            resolver: async (parent, args, context, info, extra = {}) => {
+                const { data: { dv, sender, token } } = args
+                const userData = {
+                    dv,
+                    sender,
+                    type: 'resident',
+                    isPhoneVerified: false,
+                }
+                const [action] = await ConfirmPhoneActionServerUtils.getAll(context,
+                    {
+                        token,
+                        expiresAt_gte: new Date().toISOString(),
+                        completedAt: null,
+                        isPhoneVerified: true,
+                    }
+                )
+                if (!action) {
+                    throw new Error(`${CONFIRM_PHONE_ACTION_EXPIRED}] Unable to find confirm phone action`)
+                }
+                userData.phone = action.phone
+                userData.type = 'resident'
+                userData.isPhoneVerified = true
+                userData.password = getRandomString()
+                let [user] = await UserServerUtils.getAll(context, { type: 'resident', phone: userData.phone })
```

we should throw an exception if we get more than one object